experimental/oak_baremetal_channel/src/frame.rs

```diff
@@ -0,0 +1,161 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+extern crate alloc;
+
+use alloc::{format, vec, vec::Vec};
+use bitflags::bitflags;
+use ciborium_io::{Read, Write};
+
+pub const PADDING_SIZE: usize = 4;
+
+type Length = u16;
+pub const LENGTH_SIZE: usize = 2;
+static_assertions::assert_eq_size!([u8; LENGTH_SIZE], Length);
+
+bitflags! {
+    #[derive(Default)]
+    pub struct Flags: u16 {
+        const START = 1;
+        const END = 2;
```

Comment:
Could you add links to the spec for this and most other structs / fields / offsets?

Reply:
Added some references that link the structs to the spec. :)

I'm not sure if it makes sense to go too far into details and link to individual lines, I'd be worried it would cause a lot of needless overhead when updating the spec